### PR TITLE
Iteration 5 for File detail & new sharing - Comments

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Based on #2685 

TODO
* [x] make submit button nicer https://github.com/nextcloud/android/pull/2690/commits/c2c338a6d5f071fdd0133f244474436e14c3dabd
* [x] tint underline color https://github.com/nextcloud/android/pull/2690/commits/d302e3f0d8c078567e7529071781455bb7f624c6
* [x]  #2685 merged + rebased on master afterwards

- enter a new comment
	- refreshes activity stream
	- or shows snackbar with error

device-2018-06-06-112600

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
I'll take care of the button 👍

File: src/main/java/com/owncloud/android/ui/fragment/FileFragment.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
         Bundle bundle = getArguments();
-        setFile((OCFile) bundle.getParcelable(EXTRA_FILE));
+        setFile(bundle.getParcelable(EXTRA_FILE));
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/FileFragment.java]: I accidentally touched the file and now AndroidStudio always tries to enhance this…

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky button has been beautified, see https://github.com/nextcloud/android/pull/2690/commits/c2c338a6d5f071fdd0133f244474436e14c3dabd

**Comment by AndyScherzinger** [comment]
Edit Text tinting has been added via https://github.com/nextcloud/android/pull/2690/commits/d302e3f0d8c078567e7529071781455bb7f624c6

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky except for the rebase/conflict-resolving is there anything left open for this PR since I did the 2 remaining TODOs you wrote down?

**Comment by tobiasKaminsky** [comment]
Same as the other PR, once this is ready an versions PR is merged, we need do rebase this.

**Comment by tobiasKaminsky** [comment]
This is now ready for review :+1:

**Comment by AndyScherzinger** [comment]
:+1: tested and works fine for me :+1:

**Comment by mario** [comment]
Conflicts + submit button should be rotated by 45 degrees to the left.

**Comment by AndyScherzinger** [comment]
> submit button should be rotated by 45 degrees to the left.

isn't Talk the only app doing this? :stuck_out_tongue_closed_eyes:  (looking at mail/messengers/etc.)

**Comment by mario** [comment]
@AndyScherzinger yes. Consistency :P

File: src/main/java/com/owncloud/android/operations/CommentFileOperation.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            Log.e(TAG, "Post comment to file with id " + fileId + " failed: " + result.getLogMessage(), e);
+        }
+
+        return result;
+    }
+
+    private boolean isSuccess(int status) {
+        return status == HttpStatus.SC_CREATED;
+    }
+}
```
Comment (mario) [inline src/main/java/com/owncloud/android/operations/CommentFileOperation.java]: New line.

File: src/main/java/com/owncloud/android/operations/CommentFileOperation.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
+/*
```
Comment (mario) [inline src/main/java/com/owncloud/android/operations/CommentFileOperation.java]: Whole file should be in a library.

Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/operations/CommentFileOperation.java]: General issue: ATM all "operations" are afaik _not_ in the lib but the app...

**Comment by tobiasKaminsky** [comment]
>> submit button should be rotated by 45 degrees to the left.
>
>isn't Talk the only app doing this? :stuck_out_tongue_closed_eyes:  (looking at mail/messengers/etc.)

Having a look at material icons (https://material.io/tools/icons/?icon=send&style=baseline)
image

and our web ui:
image

We should rather change the icon in talk, to stay consistent.
@nextcloud/designers

File: src/main/java/com/owncloud/android/operations/CommentFileOperation.java (hunk 3 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/operations/CommentFileOperation.java]: Currently this cannot be moved this easily as these *Operation extends RemoteOperation which is a class in files app.

I guess you confuse it with *RemoteOperations, e.g.:

RemoveFileOperation is in Files app (handles app specific stuff)
RemoveRemoteFileOperation is in library (and handles the real webdav call)

**Comment by tobiasKaminsky** [comment]
! Rebased

**Comment by jancborchardt** [comment]
@tobiasKaminsky yup – the icon you show is correct. Looking at Signal they do the same.

- The first line of the comment »Commented by you« is not really useful. It should simply say »You« instead
- Similar for the icon: Instead of the generic comment icon, it should show your avatar
- And the comment text below should have the same text size as the normal text. We shouldn’t have too many different text sizes as that looks strange.
- We can remove the »Today« header – it’s not necessary to show that one. Should only start with »Yesterday«.

Also cc @nickvergessen cause I guess these things have to do with notifications?

**Comment by AndyScherzinger** [comment]
> The first line of the comment »Commented by you« is not really useful. It should simply say »You« instead

That is the content we get from the server, so @nickvergessen or somebody else would have to change that in the server.

**Comment by mario** [comment]
👍

**Comment by tobiasKaminsky** [comment]
@jancborchardt this is all from server side, except of the date header "today, yesterday, …", which is the same as the activity stream.

Please do not merge this, as this would then be merged in versions branch.

**Comment by AndyScherzinger** [comment]
> Please do not merge this, as this would then be merged in versions branch.

@tobiasKaminsky you rebased the branch and versions has been merged, so I'd say the PR can be merged now, right? :)

**Comment by jancborchardt** [comment]
@tobiasKaminsky ok, will open in Activity then.

But text size is from Android also, right? Can you adjust that as I mentioned above?

**Comment by tobiasKaminsky** [comment]
@jancborchardt I will do it in a following PR.

**Pull request merged**